clippy_lints/src/types.rs

```diff
@@ -721,6 +734,21 @@ fn is_any_trait(t: &hir::Ty<'_>) -> bool {
     false
 }
 
+fn get_bounds_if_impl_trait<'tcx>(cx: &LateContext<'tcx>, qpath: &QPath<'_>, id: HirId) -> Option<GenericBounds<'tcx>> {
```

Comment:
A typo, it seems?
```suggestion
fn get_bounds_of_impl_trait<'tcx>(cx: &LateContext<'tcx>, qpath: &QPath<'_>, id: HirId) -> Option<GenericBounds<'tcx>> {
```

Reply:
I'm afraid not because it returns `None` if the `qpath` is not an impl trait, and actually checking whether the `qpath` is an impl trait is one of the main role of the function. I named it so hoping it looks similar to [`get_if_local`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/hir/map/struct.Map.html#method.get_if_local). If it's confusing, something like `get_bounds_provided_impl_trait` might be more suitable?